realm/realm-library/src/androidTest/java/io/realm/BulkInsertTests.java

```diff
@@ -634,4 +634,18 @@ public void linkingManagedToUnmanagedObject() {
         assertEquals(42, first.getFieldObject().getFieldLong());
         assertEquals(2, realm.where(AllJavaTypes.class).count());
     }
+
+    @Test
```

Comment:
IMO we should also add a test for what actually triggered this issue, namely that lists are no longer cleared